### PR TITLE
Add a new field to SignalR Resource property

### PR DESCRIPTION
**Pull request by juniwang**
### Changes:
 - Add new property `features` to `SignalRResource`, which will used to query or update the FeatureFlags of SignalR resource. Corresponding examples updated.
- remove `enum` property of `api-version`. The extra enum will fail .NET SDK generation
- Add `x-ms-long-running-operation-options` for long runnint operations.
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5823 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5823/commits/5ebc9368b3e6d85e967e7feaa3edfae80a8b5f56

**Comment by dsgouda** [comment]
@juniwang Please fix the merge conflicts

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
stable version but additive changes, don't need ARM approval.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Will approve once merge conflicts are resolved.

**Comment by juniwang** [comment]
> Will approve once merge conflicts are resolved.
@dsgouda Conflicts resolved

**Comment by dsgouda** [comment]
@shahabhijeet @NelsonDaniel Please approve

**Review by dsgouda — CHANGES_REQUESTED** [review]
@juniwang the diffs here are hard to read, can you pull latest changes from master and reapply your chanes

**Comment by juniwang** [comment]
@dsgouda Thanks. Resetted using master branch and applid changes.

**Review by NelsonDaniel — APPROVED** [review]

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: `features` of `SignalRCreateOrUpdateProperties` could be `null` or empty array. But if the array is not empty, both `name` and `value` are required(as line 910 suggests) for each item in the array.

**Comment by juniwang** [comment]
@dsgouda @shahabhijeet Could you have a review again?

**Comment by dsgouda** [comment]
@shahabhijeet PTAL

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (dsgouda) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: A better way to word the description would be
"FeatureFlags that are not included in the parameters for the update operation will not be affected/modified."
@shahabhijeet thoughts?

**Comment by juniwang** [comment]
Hi @shahabhijeet, Could you please have another review? It seems that the PR is blocked by you

**Review by dsgouda — APPROVED** [review]

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (dsgouda) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: nit: This doesn't need to be set explicitly

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (dsgouda) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: The description is a bit confusing since there can only be a limited number of values. `Type/kind of feature` makes more sense?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a couple of comments

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (shahabhijeet) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: @juniwang any reason you are adding this to PUT operations? I was under the impression that this is only for POST.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (shahabhijeet) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: @juniwang same as above, why specify for delete?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Telling from [this doc](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation-options): **When x-ms-long-running-operation is specified, there should also be a x-ms-long-running-operation-options specified.**. 

Will remove it.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Removed. This produces a warning:
```
WARNING (AvoidNestedProperties/R2001/SDKViolation): Consider using x-ms-client-flatten to provide a better end user experience
    - file:///mnt/d/workspace/github/juniwang/azure-rest-api-specs/specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json:924:8 ($.definitions.SignalRFeature.properties.properties)
```
If it doens't matter, we may suppress the warning in future.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Removed

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Removed

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Updated

**Comment by juniwang** [comment]
@shahabhijeet @dsgouda spec updated to rosolve comments

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (dsgouda) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: I see, we discourage too many levels of nesting especially on input parameters hence the warning. I don't think suppressing this warning is a good idea, please consider modelling the service in a way that it does not have nesting in properties like so. 
Not a blocker

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (dsgouda) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: I don't see any changes, not blocker though

**Review by dsgouda — APPROVED** [review]

**Comment by juniwang** [comment]
@shahabhijeet PTAL

**Comment by juniwang** [comment]
@shahabhijeet any other comments? Or can someone review again on behalf of Abhijeet?

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
